contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {EnumerableSet} from "../../../../shared/vendor/openzeppelin-solidity/v.4.8.0/contracts/utils/structs/EnumerableSet.sol";
+import {ITermsOfServiceAllowList} from "./interfaces/ITermsOfServiceAllowList.sol";
+import {Routable, ITypeAndVersion} from "../Routable.sol";
+import {IOwnable} from "../../../../shared/interfaces/IOwnable.sol";
+
+/**
+ * @notice A contract to handle access control of subscription management dependent on signing a Terms of Service
+ */
+
+contract TermsOfServiceAllowList is Routable, ITermsOfServiceAllowList {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  EnumerableSet.AddressSet private s_allowedSenders;
+  EnumerableSet.AddressSet private s_blockedSenders;
+
+  error InvalidProof();
+  error RecipientIsBlocked();
+  // ================================================================
+  // |                     Configuration state                      |
+  // ================================================================
+  struct Config {
+    bool enabled;
+    address proofSignerPublicKey;
+  }
+  Config private s_config;
+  event ConfigSet(bool enabled);
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(address router, bytes memory config) Routable(router, config) {}
+
+  // ================================================================
+  // |                    Configuration methods                     |
+  // ================================================================
+  /**
+   * @notice Sets the configuration
+   * @param config bytes of config data to set the following:
+   *  - enabled: boolean representing if the allow list is active, when disabled all usage will be allowed
+   *  - proofSignerPublicKey: public key of the signer of the proof
+   */
+  function _setConfig(bytes memory config) internal override {
```

Comment:
Why does Routable require this generic _setConfig() method? Can't we just allow all contracts to have their own config method with all needed params? Is config setting supposed to always go through the router?

Reply:
Yes this is the internal logic for a function that always gets called by the Router

Reply:
Yeah but why? What's wrong with just setting the config on the destination contract directly?